src/ddspdrum/module.py

```diff
@@ -235,3 +236,166 @@ def play(self):
         amp = self.control_to_sample_rate(self.get_envelope())
         signal = self.fix_length(self.get_audio(), len(amp))
         return amp * signal
+
+
+class SVF(SynthModule):
+    """
+    A State Variable Filter that can do lowpass, highpass, bandpass, and bandreject
+    filtering. Allows modulation of the cutoff frequency and an adjustable resonance
+    parameter. Can self-oscillate to make a sine / cosine wave.
+    """
+
+    def __init__(
+        self,
+        mode: str = "LPF",
```

Comment:
One thing I would like to avoid---as much as possible if it doesn't make us crazy---is having public modules that have categorical parameters ("LPF", "BPF", and "BSF"). My preference would be that we have a semiprivate `_SVF` module but public implementations are called `LPF` and `BPF` and `BSF`.

The idea is that backpropagating through a categorical variable is hard whereas, if there is an option to pickone of the three, we just have all three modules side by side, we use softmax to have an amplitude probability distribution go through the three (summing to one), and then potentially penalize the amplitude if its not in one-on two-off configuration.